### PR TITLE
Karhunen loeve

### PR DESCRIPTION
**Pull request by regislebrun**
The main purpose of this PR is to provide algorithms to perform Karhunen Loeve decompositions of covariance models and to give access to projectors on the resulting functional basis.

It also provides some bug fixes in FunctionalChaosRandomVector, distribution comparrison, Histogram, doc and in various comments.

It adds properties to processes such as an accessor for the trend and the covariance model for other prossesses than the TemporalNormalProcess.

It adds marginal extraction for Field and ProcessSample

Finally, it provides a new scale management in TensorizedCovarianceModel, mandatory to allow the extraction of the covariance model of AggregatedProcess when the individual processes have different scales.

### CODE REVIEW

File: lib/src/Uncertainty/Process/KarhunenLoeveQuadratureAlgorithm.cxx [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+
+static const Factory<KarhunenLoeveQuadratureAlgorithm> Algorithm_KarhunenLoeveQuadratureAlgorithm;
+
+/* Constructor without parameters */
+KarhunenLoeveQuadratureAlgorithm::KarhunenLoeveQuadratureAlgorithm()
+  : KarhunenLoeveAlgorithmImplementation()
+  , domain_()
+  , basis_()
+  , basisSize_(0)
+  , experiment_()
```
Comment (jschueller) [inline lib/src/Uncertainty/Process/KarhunenLoeveQuadratureAlgorithm.cxx]: KarhunenLoeveQuadratureAlgorithm.cxx:54:5: warning: field 'basisSize_' will be initialized after field 'experiment_' [-Wreorder]

File: python/doc/pyplots/KarhunenLoeveQuadratureFactory.py [hunk 2]
```diff
@@ -6,14 +6,16 @@
...
+experiment = ot.GaussProductExperiment(basis.getMeasure(), [20])
 mustScale = False
 threshold = 0.001
 factory = ot.KarhunenLoeveQuadratureFactory(
     domain, experiment, basis, basisSize, mustScale, threshold)
 model = ot.AbsoluteExponential([1.0])
 ev = ot.NumericalPoint()
 functions = factory.build(model, ev)
+print "ev=", ev
+print "functions=", functions
```
Comment (jschueller) [inline python/doc/pyplots/KarhunenLoeveQuadratureFactory.py]: these should be removed, or at least use print as a function:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/matplotlib/sphinxext/plot_directive.py", line 517, in run_code
    six.exec_(code, ns)
  File "<string>", line 17
    print "ev=", ev
              ^
SyntaxError: Missing parentheses in call to 'print'
```

File: lib/src/Uncertainty/Process/CMakeLists.txt [hunk 3]
```diff
@@ -39,3 +40,4 @@ ot_install_header_file (WhittleFactoryState.hxx)
 ot_install_header_file (ARMALikelihoodFactory.hxx)
 ot_install_header_file (ConditionedNormalProcess.hxx)
 ot_install_header_file (KarhunenLoeveQuadratureFactory.hxx)
+#ot_install_header_file (KarhunenLoeveQuadratureAlgorithm.hxx)
```
Comment (jschueller) [inline lib/src/Uncertainty/Process/CMakeLists.txt]: why is this commented ?

**Comment by jschueller** [comment]
This needs rebasing.

File: lib/src/Uncertainty/Distribution/Mixture.cxx [hunk 4]
```diff
@@ -603,6 +603,25 @@ NumericalSample Mixture::getSupport(const Interval & interval) const
...
+  // Aggregate all the singularities of the atoms including the bounds of the range
+  for (UnsignedInteger i = 0; i < distributionCollection_.getSize(); ++i)
+    {
+      singularities.add(distributionCollection_[i].getRange().getLowerBound()[0]);
+      singularities.add(distributionCollection_[i].getSingularities());
+      singularities.add(distributionCollection_[i].getRange().getUpperBound()[0]);
+    }
+  // Sort the singularities
+  std::sort(singularities.begin(), singularities.end());
+  NumericalPoint result(singularities.getDimension() - 2);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/Mixture.cxx]: Why do you put aside the first and last ones?
We way also face to duplicates....

**Comment by sofianehaddad** [comment]
I think you can squash 	41e759a,  74aaa00,  650150e

**Pull request merged**